Library/PushRegistrationType.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import ReactiveSwift
+import Result
+import UIKit
+import UserNotifications
+
+public enum PushNotificationType: CaseIterable {
```

Comment:
An `enum` that allows us to not care about `UIUserNotificationType` or `UNAuthorizationOptions` at the callsite.